examples/classification/benchmark.py

```diff
@@ -0,0 +1,393 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+An example script to benchmark a classification model
+
+Supports ONNX files or SparseZoo stubs for models
+Supports benchmarking on random or provided data
+Supports overriding original input shape
+Supports both ONNX and DeepSparse runtimes
+
+#########
+Command help:
+usage: benchmark.py [-h] [-e {deepsparse,onnxruntime}] [--data-path DATA_PATH]
+                    [--image-shape IMAGE_SHAPE [IMAGE_SHAPE ...]]
+                    [-b BATCH_SIZE] [-c NUM_CORES] [-s NUM_SOCKETS]
+                    [-i NUM_ITERATIONS] [-w NUM_WARMUP_ITERATIONS]
+                    model_filepath
+
+Benchmark classification model
+
+positional arguments:
+  model_filepath        The full filepath of the ONNX model file or SparseZoo
+                        stub to the model for DeepSparse or ONNX runtime
+                        benchmarks.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -e {deepsparse,onnxruntime}, --engine {deepsparse,onnxruntime}
+                        Inference engine backend to run benchmark on. Choices
+                        are 'deepsparse', and 'onnxruntime'. Default is
+                        'deepsparse'
+  --data-path DATA_PATH
+                        path to SparseZoo stub for data sample to benchmark
+                        with, or to directory of .npz files with named sample
+                        inputs to the given model, inputs should be for a
+                        single sample with no batch dimension included.
+                        Defaults to benchmark on random image samples with
+                        channel size 3.
+  --image-shape IMAGE_SHAPE [IMAGE_SHAPE ...]
+                        Image shape to benchmark with, must be two integers.
+                        If None model is used to infer image shape
+  -b BATCH_SIZE, --batch-size BATCH_SIZE
+                        The batch size to run the benchmark for
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the benchmark on,
+                        defaults to None where it uses all physical cores
+                        available on the system. For DeepSparse benchmarks,
+                        this value is the number of cores per socket
+  -s NUM_SOCKETS, --num-sockets NUM_SOCKETS
+                        For DeepSparse benchmarks only. The number of physical
+                        cores to run the benchmark on. Defaults to None where
+                        is uses all sockets available on the system
+  -i NUM_ITERATIONS, --num-iterations NUM_ITERATIONS
+                        The number of iterations the benchmark will be run for
+  -w NUM_WARMUP_ITERATIONS, --num-warmup-iterations NUM_WARMUP_ITERATIONS
+                        The number of warmup iterations that will be executed
+                        before the actual benchmarking
+
```

Comment:
let's add an example of invoking the script

Reply:
👌